jung-visualization/src/main/java/edu/uci/ics/jung/visualization/transform/HyperbolicTransformer.java

```diff
@@ -8,11 +8,13 @@
  */
 package edu.uci.ics.jung.visualization.transform;
 
-import edu.uci.ics.jung.layout.model.PointModel;
+import static edu.uci.ics.jung.visualization.layout.AWT.POINT_MODEL;
+
 import edu.uci.ics.jung.layout.model.PolarPoint;
-import edu.uci.ics.jung.visualization.layout.AWTPointModel;
-import java.awt.Component;
+import java.awt.*;
 import java.awt.geom.Point2D;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * HyperbolicTransformer wraps a MutableAffineTransformer and modifies the transform and
```

Comment:
Do these  `*Transformer` classes need to have a hard-wired PointModel, or can they also be generic in P extends PointModel?

Reply:
Do they use Rectangle and Shape?

Reply:
HyperbolicTransformer doesn't.  LensTransformer (which this extends) does.  I assume that this means that this would depend on us providing a RectangleModel interface, etc.

We can revisit this in a follow-up change; I'm just still trying to get a handle on the design.

Reply:
The design of the java2d direct rendering code is not much different that it has been for years. I am just cleaning things up, fixing bugs, and documenting it. It has a more limited shelf life than something with javafx or a more modern scene graph based rendering model, which will make much of the current 2d stuff obsolete, since the scene graph model will already handle transformations, picking etc.